crates/biome_js_type_info/src/type_info/literal.rs

```diff
@@ -0,0 +1,213 @@
+        Self(Text::Owned(value))
+    }
+}
+
+impl From<&str> for StringLiteral {
+    fn from(value: &str) -> Self {
+        Self(Text::Owned(value.to_string()))
+    }
+}
+
+impl From<StringLiteral> for Text {
+    fn from(value: StringLiteral) -> Self {
+        value.0
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn parse_boolean() {
+        assert_eq!(BooleanLiteral::parse("true"), Some(BooleanLiteral(true)));
+        assert_eq!(BooleanLiteral::parse("True"), Some(BooleanLiteral(true)));
+        assert_eq!(BooleanLiteral::parse("TRUE"), Some(BooleanLiteral(true)));
+        assert_eq!(BooleanLiteral::parse("false"), Some(BooleanLiteral(false)));
+        assert_eq!(BooleanLiteral::parse("False"), Some(BooleanLiteral(false)));
+        assert_eq!(BooleanLiteral::parse("FALSE"), Some(BooleanLiteral(false)));
+    }
+
+    #[test]
+    fn parse_number_basic() {
+        assert_eq!(
+            NumberLiteral::parse("1234567890"),
+            Some(NumberLiteral(1_234_567_890.0))
+        );
+        assert_eq!(NumberLiteral::parse("42"), Some(NumberLiteral(42.0)));
+        assert_eq!(NumberLiteral::parse("0"), Some(NumberLiteral(0.0)));
+    }
+
+    #[test]
+    fn parse_number_legacy_octal() {
+        assert_eq!(NumberLiteral::parse("0888"), Some(NumberLiteral(888.0)));
+        assert_eq!(NumberLiteral::parse("0777"), Some(NumberLiteral(511.0)));
+    }
+
+    #[test]
+    fn parse_number_exponential() {
+        assert_eq!(NumberLiteral::parse("0e-5"), Some(NumberLiteral(0.0)));
+        assert_eq!(NumberLiteral::parse("0e+5"), Some(NumberLiteral(0.0)));
+        assert_eq!(NumberLiteral::parse("5e1"), Some(NumberLiteral(50.0)));
+        assert_eq!(NumberLiteral::parse("175e-2"), Some(NumberLiteral(1.75)));
+        assert_eq!(NumberLiteral::parse("1e3"), Some(NumberLiteral(1000.0)));
+        assert_eq!(NumberLiteral::parse("1e-3"), Some(NumberLiteral(0.001)));
+        assert_eq!(NumberLiteral::parse("1E3"), Some(NumberLiteral(1000.0)));
+    }
+
+    #[test]
+    fn parse_number_binary() {
+        assert_eq!(
+            NumberLiteral::parse("0b10000000000000000000000000000000"),
+            Some(NumberLiteral(2_147_483_648.0))
+        );
+        assert_eq!(
+            NumberLiteral::parse("0b01111111100000000000000000000000"),
+            Some(NumberLiteral(2_139_095_040.0))
+        );
+        assert_eq!(
+            NumberLiteral::parse("0B00000000011111111111111111111111"),
+            Some(NumberLiteral(8_388_607.0))
+        );
+    }
+
+    #[test]
+    fn parse_number_octal() {
+        assert_eq!(NumberLiteral::parse("0O755"), Some(NumberLiteral(493.0)));
+        assert_eq!(NumberLiteral::parse("0o644"), Some(NumberLiteral(420.0)));
+    }
+
+    #[test]
+    fn parse_number_hexadecimal() {
+        // assert_eq!(
+        //     NumberLiteral::parse("0xFFFFFFFFFFFFFFFFF"),
+        //     Some(NumberLiteral(295147905179352830000.0))
+        // );
```

Comment:
Why is this one commented? TODO?

Reply:
It taks 68 bits and can't parse it using `u64::from_str_radix`. Left a comment and marked as TODO.